### ISSUE TITLE
Exclude absent classes from categorical_tversky loss computation

### ISSUE THREAD

**Issue by klimesm** [ctu-geoforall-lab/cnn-lib#47]
`categorical_tversky` averages tversky score over all classes including
those not present in the batch. If a class has no ground-truth pixels,
its tversky score is 0, which incorrectly lowers the average loss.

This is especially problematic for imbalanced datasets where some
classes (e.g. cloud-shadow) are rare and may not appear in every batch.

## Proposed fix

Average only over classes that have at least one ground-truth pixel
in the batch:

```python
gt_per_class = tf.reduce_sum(
    tf.reduce_sum(true_pos, axis=(1, 2)) + tf.reduce_sum(false_neg, axis=(1, 2)),
    axis=0
)
class_present = tf.cast(gt_per_class > 0, tf.float32)

weighted_tversky = weight_tensor * tversky * class_present
num_present = tf.reduce_sum(class_present)
loss = 1 - tf.reduce_sum(weighted_tversky) / (num_present + 1e-7)
```

## Note

This is a behaviour change — it will affect all trainings with tversky/dice loss.
Thus I would adress this after thesis submission.
